executor/distsql.go

```diff
@@ -750,9 +749,10 @@ func (e *IndexLookUpExecutor) Next(ctx context.Context, req *chunk.Chunk) error
 		if resultTask == nil {
 			return nil
 		}
-		if resultTask.cursor < len(resultTask.rows) {
-			numToAppend := mathutil.Min(len(resultTask.rows)-resultTask.cursor, req.RequiredRows()-req.NumRows())
-			req.AppendRows(resultTask.rows[resultTask.cursor : resultTask.cursor+numToAppend])
+		rows := resultTask.getRows()
+		if resultTask.cursor < len(rows) {
+			numToAppend := mathutil.Min(len(rows)-resultTask.cursor, req.RequiredRows()-req.NumRows())
+			req.AppendRows(rows[resultTask.cursor : resultTask.cursor+numToAppend])
```

Comment:
IMO, there is no need to copy rows using `getRows`. Could you explain that to me?

Reply:
Need an explaination @likzn

Reply:
I see, you can not use pdqsort without changing the definition. Then we may need some microbenchmarks to test if really improves.

Because pdqsort may be slower if an extra copy is needed.